diff -up vic-20071009svn/vic/tcl/ui-stats.tcl.mtrace vic-20071009svn/vic/tcl/ui-stats.tcl
--- vic/tcl/ui-stats.tcl.mtrace	2007-08-08 12:50:46.000000000 +1000
+++ vic/tcl/ui-stats.tcl	2007-12-03 08:28:25.000000000 +1000
@@ -698,9 +698,9 @@ proc create_mtrace_window {src dir} {
 
 	global V
 	if {$dir=="to"} {
-		set cmd "|mtrace [$V(data-net) interface] [$src addr] [$V(data-net) addr]"
+		set cmd "|/usr/sbin/mtrace [$V(data-net) interface] [$src addr] [$V(data-net) addr]"
 	} else {
-		set cmd "|mtrace [$src addr] [$V(data-net) addr]"
+		set cmd "|/usr/sbin/mtrace [$src addr] [$V(data-net) addr]"
 	}
 	if [catch "open {$cmd} r" fd] {
 		$w.t.text insert end "mtrace error: $fd"
