--- vic/video/grabber-video4linux.cpp.videodev_h	2006-12-13 22:29:55.000000000 +1000
+++ vic/video/grabber-video4linux.cpp	2007-05-07 13:59:58.000000000 +1000
@@ -38,10 +38,10 @@
 extern "C"
 {
 #include <asm/types.h>
-//#include <linux/videodev.h>
+#include <linux/videodev.h>
 }
 
-#include "videodev.h"
+//#include "videodev.h"
 #include "grabber.h"
 #include "vic_tcl.h"
 #include "device-input.h"
