--- src/gtk/gsockgtk.cpp.orig	2007-07-05 05:25:08.000000000 +1000
+++ src/gtk/gsockgtk.cpp	2010-03-09 10:11:42.781162992 +1000
@@ -15,8 +15,13 @@
 #include <stdlib.h>
 #include <stdio.h>
 
+// newer versions of glib define its own GSocket but we unfortunately use this
+// name in our own (semi-)public header and so can't change it -- rename glib
+// one instead
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
 #include <glib.h>
+#undef GSocket
 
 #include "wx/gsocket.h"
 #include "wx/unix/gsockunx.h"
