diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' accessgrid3.2-3.2~/AccessGrid/AccessGrid/AGXMLRPCServer.py accessgrid3.2-3.2/AccessGrid/AccessGrid/AGXMLRPCServer.py
--- accessgrid3.2-3.2~/AccessGrid/AccessGrid/AGXMLRPCServer.py	2009-04-29 05:32:05.000000000 +1000
+++ accessgrid3.2-3.2/AccessGrid/AccessGrid/AGXMLRPCServer.py	2010-10-15 10:10:54.924854581 +1000
@@ -44,7 +44,13 @@
                 return (sock, addr)
             except socket.timeout:
                 pass
+            # handle_timeout was introduced in python 2.6
+            # in earlier python versions, call handle_timeout
+            #  here to react to the timeout exception above
+            if sys.version < (2,6):
+                self.handle_timeout()
 
+    def handle_timeout(self):
             try:
                 if self.callback != None:
                     self.callback()
@@ -55,7 +61,6 @@
             except:
                 traceback.print_exc()
 
-
     def run(self):
         self.running = True
         while self.running:
