--- AccessGrid-3.2/AccessGrid/AccessGrid/UrllibTransport.py	2009-04-29 05:32:05.000000000 +1000
+++ AccessGrid-3.3/AccessGrid/AccessGrid/UrllibTransport.py	2011-04-28 17:16:05.167791636 +1000
@@ -65,7 +65,10 @@
         host, extra_headers, x509 = self.get_host_info(host)
         if self.httpconn:
             self.freeconn()
-        self.httpconn = TimeoutHTTP(host,timeout=self.timeout)
+        if hasattr(xmlrpclib.Transport, 'single_request'):
+            self.httpconn = TimeoutHTTPConnection(host,timeout=self.timeout)
+        else:
+            self.httpconn = TimeoutHTTP(host,timeout=self.timeout)
         return self.httpconn
             
     def freeconn(self):
@@ -125,7 +128,10 @@
 
     def make_connection(self, host):
         host, extra_headers, x509 = self.get_host_info(host)
-        self.httpconn = TimeoutHTTP(host,timeout=self.timeout)
+        if hasattr(xmlrpclib.Transport, 'single_request'):
+            self.httpconn = TimeoutHTTPConnection(host,timeout=self.timeout)
+        else:
+            self.httpconn = TimeoutHTTP(host,timeout=self.timeout)
         return self.httpconn
         
     def freeconn(self):
