diff -urNad accessgrid3.1-3.1~/AccessGrid/AccessGrid/hosting/ZSI/Server.py accessgrid3.1-3.1/AccessGrid/AccessGrid/hosting/ZSI/Server.py
--- accessgrid3.1-3.1~/AccessGrid/AccessGrid/hosting/ZSI/Server.py	2007-04-27 01:35:09.000000000 +1000
+++ accessgrid3.1-3.1/AccessGrid/AccessGrid/hosting/ZSI/Server.py	2008-03-23 21:22:48.078319517 +1000
@@ -21,7 +21,7 @@
 
 from AccessGrid import Log
 log = Log.GetLogger(Log.Hosting)
-import select
+import select, socket
 
 def GetSOAPContext():
     return None
@@ -67,6 +67,10 @@
                 r,w,e = select.select([self._server.socket], [], [], pause)
                 if r:
                     self._server.handle_request()
+            except socket.error, ex:
+                if ex[0] == 4: # interrupted system call
+                    continue
+                log.exception("Exception in SOAP server main loop")
             except:
                 log.exception("Exception in SOAP server main loop")
                 
