diff -up ZSI-2.0/setup.py.no_egg_PyXML ZSI-2.0/setup.py
--- ZSI-2.0/setup.py.no_egg_PyXML	2006-09-28 08:55:49.000000000 +1000
+++ ZSI-2.0/setup.py	2009-02-25 10:58:51.000000000 +1000
@@ -44,7 +44,6 @@ if hasSetuptools:
             'wsdl2dispatch = ZSI.generate.commands:wsdl2dispatch',
         ],
     }
-    additional_params['install_requires'] = [ "PyXML >= 0.8.3", ]
     additional_params['setup_requires'] = [ "setuptools >= 0.6c3", ]
     additional_params['dependency_links'] = [
         "http://sourceforge.net/project/showfiles.php?group_id=6473&package_id=6541&release_id=286213",
